### PR TITLE
Should subsample Convolution1D on correct axis

### PR DESCRIPTION
**Pull request by nehz**

### CODE REVIEW

**Comment by fchollet** [comment]
Please clarify.

**Comment by nehz** [comment]
For 1D the input is reshaped as:
`X = T.reshape(X, (X.shape[0], X.shape[1], X.shape[2], 1)).dimshuffle(0, 2, 1, 3)`

**Comment by fchollet** [comment]
Can you demonstrate the correctness of your proposed change, or should I close this?

**Comment by nehz** [comment]
model = Sequential()

    model.add(Convolution1D(input_dim=1, nb_filter=1, filter_length=1, subsample_length=2, border_mode='valid'))
    model.add(Flatten())
    model.add(Dense(2, 1))
    model.compile(loss='mse', optimizer='rmsprop')

    X = np.array([[[1], [1], [1], [1]]])
    y = np.array([1])
    model.train_on_batch(X, y)

Expect subsample_length to subsample steps

Also based on how:
http://deeplearning.net/software/theano/library/tensor/nnet/conv.html#theano.tensor.nnet.conv.conv2d

and how the input is reshaped. 
There was a similar issue #509
